contributing.md

```diff
@@ -0,0 +1,155 @@
+# Contributing
+
+You can contribute to Warp by raising issues and PRs. Simply filing issues for problems you encounter is a great way to contribute. Contributing implementations is greatly appreciated.
+
+## DOs and DON'Ts
+
+Please do:
+
+- **DO** give priority to the current style of the project or file you're changing even if it diverges from the general guidelines.
+- **DO** include tests for both, new features and bug fixes
+- **DO** keep the discussions focused on the initial topic
+
+Please do not:
+
+- **DON'T** raise PRs with style changes
+- **DON'T** surprise us with big pull requests. It's better to first raise it as an issue and have it discussed
+- **DON'T** commit code that you didn't author. This may breach IP
+- **DON'T** submit PRs that alter licensing, contributing guide lines or code of conduct
+
+## Installation
+
+First, please make sure you have all the required external [dependencies](dependencies.md) installed.
+
+Then, with a Python3.9 virtual environment activated, perform the following instructions:
+
+1. Clone this repo and change directory into the `warp` folder.
+
+2. Install the JavaScript dependencies:
+
+```bash
+yarn
+```
+
+3. Install the Python dependencies:
+
+```bash
+pip install -r requirements.txt
+```
+
+If you are using a M1 chipped Mac and getting a `'gmp.h' file not found` error when installing Cairo run the following:
+
+```bash
+CFLAGS=-I`brew --prefix gmp`/include LDFLAGS=-L`brew --prefix gmp`/lib pip install ecdsa fastecdsa sympy
+```
+
+Then run the pip command above again.
+
+4. Compile the project:
+
+```bash
+yarn tsc
+```
+
+5. Compile Warp librariesÑ
+
+```bash
+yarn warplib
+```
+
+6. Test the installation worked by transpiling an example ERC20 contract:
+
+```bash
+bin/warp transpile example_contracts/ERC20.sol
+```
+
+## First steps :feet:
+
+To look what features we are currently working on or tasks that are pending to do, please checkout the [project on github](https://github.com/orgs/NethermindEth/projects/30/views/3).
+
+## Developing tips :honey_pot:
+
+While developing your code, remembering to compile the project every time some minor changes are applied can be annoying. You can start a process to recompile Warp on file changes.
+
+In a separate terminal execute:
+
+```bash
+yarn dev
+```
+
+You can also run warp instructions using docker to achieve the same. This is an example using `transpile` command:
```

Comment:
It's not a property of docker, ts-node may be use regardless of environment

Reply:
I am not sure how to write what you mean. Could you add a suggestion?

Reply:
Sure, I did below